src/werkzeug/formparser.py

```diff
@@ -440,209 +357,79 @@ def get_part_charset(self, headers: Headers) -> str:
         return self.charset
 
     def start_file_streaming(
-        self, filename: str, headers: Headers, total_content_length: int
-    ) -> t.Tuple[str, t.BinaryIO]:
-        if isinstance(filename, bytes):
-            filename = filename.decode(self.charset, self.errors)
-
-        filename = self._fix_ie_filename(filename)
-        content_type = headers.get("content-type")
+        self, event: File, total_content_length: int
+    ) -> t.BinaryIO:
+        content_type = event.headers.get("content-type")
 
         try:
-            content_length = int(headers["content-length"])
+            content_length = int(event.headers["content-length"])
         except (KeyError, ValueError):
             content_length = 0
 
         container = self.stream_factory(
             total_content_length=total_content_length,
-            filename=filename,
+            filename=event.filename,
             content_type=content_type,
             content_length=content_length,
         )
-        return filename, container
+        return container
 
-    def in_memory_threshold_reached(self, size: int) -> None:
-        raise exceptions.RequestEntityTooLarge()
+    def parse(
+        self, stream: t.BinaryIO, boundary: bytes, content_length: int
+    ) -> t.Tuple[MultiDict, MultiDict]:
+        container: t.Union[t.BinaryIO, t.List[bytes]]
+        _write: t.Callable[[bytes], t.Any]
 
-    def parse_lines(
-        self,
-        file: t.BinaryIO,
-        boundary: bytes,
-        content_length: int,
-        cap_at_buffer: bool = True,
-    ) -> t.Iterator[
-        t.Tuple[
-            str, t.Union[t.Tuple[Headers, str], t.Tuple[Headers, str, str], bytes, None]
-        ]
-    ]:
-        """Generate parts of
-        ``('begin_form', (headers, name))``
-        ``('begin_file', (headers, name, filename))``
-        ``('cont', bytes)``
-        ``('end', None)``
-
-        Always obeys the grammar
-        parts = ( begin_form cont* end |
-                  begin_file cont* end )*
-        """
-        next_part = b"--" + boundary
-        last_part = next_part + b"--"
         iterator = chain(
             make_line_iter(
```

Comment:
#875 indicates that `make_line_iter` is very inefficient for binary files. Seems this could be replaced now that we can feed data in to the parser. Maybe read `buffer_size` from the stream in a while loop instead of lines in a for loop.

Reply:
Done, used `_make_chunk_iter`